pkg/resource/interfaces.go

```diff
@@ -114,6 +121,12 @@ type CompositeResourceReferencer interface {
 	GetResourceReference() *corev1.ObjectReference
 }
 
+// A UserCounter can count how many users it has.
+type UserCounter interface {
+	SetUsers(i *int64)
+	GetUsers() *int64
```

Comment:
```suggestion
	SetUsers(i int64)
	GetUsers() int64
```

I think `0` as default value makes sense for this information. How does the following sound for users field:
```go
// A ProviderConfigStatus defines the observed status of a ProviderConfig.
type ProviderConfigStatus struct {
	// Users of this provider configuration.
	Users int64 `json:"users,omitempty"`
}
```

Reply:
Makes sense. I forget why I switched it to a pointer.